catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/SourceOperations.java

```diff
@@ -442,25 +440,26 @@ private void addCatalogSourceDescriptor(Set<SourceDescriptor> descriptors) {
     // But when a local catalog provider is configured, include its content types in the
     // local site info.
     if (descriptors != null) {
-      Source cachedSource = null;
       if (catalog != null) {
-        // check the poller for cached information
-        if (frameworkProperties.getSourcePoller() != null
-            && frameworkProperties.getSourcePoller().getCachedSource(catalog) != null) {
-          cachedSource = frameworkProperties.getSourcePoller().getCachedSource(catalog);
+
+        final Set<ContentType> cachedContentTypes;
```

Comment:
❓Would this work instead?
```suggestion
        final Set<ContentType> cachedContentTypes =
          contentTypesCache.getCachedValueForSource(catalog)
          .orElseGet(() -> {
            LOGGER.debug("Unknown ContentTypes for the local catalog");
            return Collections.emptySet();
          });
```

Reply:
See https://github.com/codice/ddf/pull/3858#discussion_r253680191

Reply:
See https://github.com/codice/ddf/pull/3858/files#r253930138 😉